src/gui/spatial/index.js

```diff
@@ -35,10 +37,80 @@ realityEditor.gui.spatial.clearSpatialList = function (){
 };
 realityEditor.gui.spatial.lineAnimationList = {};
 
+
+
+
+realityEditor.gui.spatial.checkState = function() {
+    let counter = 0;
+    let counter2 = 0;
+    for (let ip in globalStates.spatial.whereIs) {
```

Comment:
Since you only care whether counter and counter2 are greater than 0 this code can be significantly optimized by replacing them with booleans historianOn and spatialOn then modifying all these for loops as follows:
```javascript
let spatialOn = false;
let historianOn = false;
for (let ip in globalStates.spatial.whereIs) {
  if (Object.keys(globalStates.spatial.whereIs[ip]).length > 0) {
    spatialOn = true;
    break;
  }
}

if (!spatialOn) {
  for (let ip in globalStates.spatial.howFarIs) {
    if (Object.keys(globalStates.spatial.howFarIs[ip]).length > 0) {
      spatialOn = true;
      break;
    }
  }
}
```